plone/app/layout/navigation/navtree.py

```diff
@@ -354,3 +364,153 @@ def insertElement(itemPaths, item, forceInsert=False):
+        return self._navtree_path
+
+    @property
+    def navtree_depth(self):
+        return self.settings.navigation_depth
+
+    @property
+    def enableDesc(self):
+        return True
+
+    @property
+    def navtree(self):
+
+        if self._navtree is not None:
+            return self._navtree
+
+        generate_tabs = self.settings.generate_tabs
+        types = self.settings.displayed_types
+        default_language = self.language_settings.default_language
+
+        request = getRequest()
+        lang_current = request.get('LANGUAGE', None) or \
+            (self.context and aq_inner(self.context).Language()) \
+            or default_language
+
+        ret = {}
+
+        # Get current object path for later determine if it's current
+        context_physical_path = self.context.getPhysicalPath()
+        if utils.isDefaultPage(self.context, self.request):
+            context_physical_path = context_physical_path[:-1]
+        context_path = '/'.join(context_physical_path)
+
+        if generate_tabs:
+            query = {
+                'path': {'query': self.navtree_path,
+                         'depth': self.navtree_depth},
+                'portal_type': {'query': types},
+                'exclude_from_nav': False,
+                'Language': lang_current,
+                'sort_on': 'getObjPositionInParent',
+            }
+            portal_catalog = getToolByName(self.context, 'portal_catalog')
+
+            res = portal_catalog.searchResults(**query)
+
+            for it in res:
+                if context_path is not None:
+                    # Determine if it's current object
+                    is_current = context_path == it.getPath()
+                pathkey = '/'.join(it.getPath().split('/')[:-1])
+                entry = {
+                    'id': it.id,
+                    'uid': it.UID,
+                    'url': it.getURL(),
+                    'title': it.Title,
+                    'review_state': it.review_state,
+                    'is_current': is_current,
+                }
+                if pathkey in ret:
+                    ret[pathkey].append(entry)
+                else:
+                    ret[pathkey] = [entry]
+
+            self._navtree = ret
+        else:
+            portal_tabs_view = getMultiAdapter((self.context, self.request),
+                                               name='portal_tabs_view')
+            self.portal_tabs = portal_tabs_view.topLevelTabs()
+            res = self.portal_tabs
+
+            for it in res:
+                if context_path is not None:
+                    # Determine if it's current object
+                    is_current = context_path == it.getPath()
+                pathkey = self.navtree_path
+                entry = {
+                    'id': it['id'],
+                    'uid': None,
+                    'url': it['url'],
+                    'title': it['title'],
```

Comment:
Please make sure the Home tab is translated! See https://github.com/plone/plone.app.layout/pull/223